### PR TITLE
Polars Struct support without unsafe blocks

### PR DESCRIPTION
**Pull request by ayax79**
Second attempt at polars Struct support. This version avoid using unsafe checks by cloning the StructArray and utilizing the into_static to convert to a StructOwned.

### CODE REVIEW

**Comment by fdncred** [comment]
cool!
/cc @sholderbach

File: crates/nu-cmd-dataframe/src/dataframe/values/nu_dataframe/conversion.rs [hunk 1]
```diff
@@ -963,10 +916,154 @@ fn series_to_values(
...
+            let fields = struct_tuple
+                .1
+                .iter()
+                .map(|f| f.name().to_string())
+                .collect();
+            Ok(Value::Record {
+                val: Record {
+                    cols: fields,
+                    vals: values?,
+                },
```
Comment (sholderbach) [inline crates/nu-cmd-dataframe/src/dataframe/values/nu_dataframe/conversion.rs:988]: In the future we will make those fields private and force you to go through either `.collect` or `Record::from_raw_cols_vals` or similar helper. 

But I can take care of updating that in a separate PR, already started a branch for those remaining uses when I came across the unsafe issue.

**Review by sholderbach — APPROVED** [review]
Thanks for reworking this. Sorry that I didn't get to it earlier.

Looks good to me and cleans up pretty nicely. Good to have the datetime logic in the helpers.

Thanks for all the tests!

**Comment by hustcer** [comment]
Thanks for adding this feature. BTW, it's found that `riscv64gc-unknown-linux-gnu` build target is broken after this PR, more detail could be found here: https://github.com/nushell/nightly/actions/runs/7231785373/job/19705342169 (I just cherry-picked 44dc89012 to tag 0.88.1). I'm wondering if there is a way to fix this? Thanks

**Comment by hustcer** [comment]
`riscv64gc-unknown-linux-gnu` build error should be fixed by: https://github.com/nushell/nushell/pull/11353

**Pull request merged**